zcash_primitives/src/primitives.rs

```diff
@@ -266,7 +330,7 @@ impl<E: JubjubEngine> Note<E> {
         // Write pk_d
         self.pk_d.write(&mut note_contents).unwrap();
 
-        assert_eq!(note_contents.len(), 32 + 32 + 8);
+        assert_eq!(note_contents.len(), 32 + 32 + 32 + 8);
```

Comment:
The extra `32` is for the asset type? Comment here would be nice.

Reply:
yes, this is the 32 bytes which corresponds to the 256 bits from the other comment, which is the jubjub point that is the asset type/value commitment generator. I think I'll change this to be broken apart (via comments) into labeled asset_type/g_d/etc like the other length assert_eq.

Reply:
Oops, its not the jubjub point (which is 382 bits, anyway...), its the arbitrary integer u32 assigned to each asset type...